modules/service_bus_subscription/main.tf

```diff
@@ -41,8 +42,14 @@ resource "azurerm_servicebus_subscription_rule" "correlation_filter" {
   filter_type     = "CorrelationFilter"
 
   correlation_filter {
-    correlation_id = var.correlation_filter.id
-    label          = var.correlation_filter.label
-    properties     = var.correlation_filter.properties
+    correlation_id      = var.correlation_filter.id != null ? var.correlation_filter.id : null
```

Comment:
I think it is better to set the default values as null instead of doing this

Reply:
Sadly the other option is not very user friendly, Basically every time I will use the module I would have to create object with all the properties, only with some of them being null
 
 
![image](https://user-images.githubusercontent.com/89966532/171350754-ecb43504-a330-4010-aad7-bf953835563c.png)

which is very uncomfortable in my opinion

Reply:
Those checks are actually not even needed. 
Also I do not think that any default value if reasonable since it will always have different values based on requirements

Reply:
I agree, good that you made it simpler! Default values (1) could be really default, like a basic SKU or default logging settings. On the other hand, you would use default = null for stuff that you do not want to set beforehand, but you still want the user to give the option to set a field or deploy a whole resource with a conditional.